math/fibonacci.c

```diff
@@ -1,23 +1,109 @@
-#include <stdio.h>
+/**
+ * @file
+ * @brief Program to print the nth term of the Fibonacci series.
+ * @details
+ * Fibonacci series generally starts from 0 and 1. Every next term in
+ * the series is equal to the sum of the two preceding terms.
+ */
 
-// Fibonnacci function
-int fib(int number)
+#include <assert.h>  /// for assert()
+#include <errno.h>  /// for errno - to determine whether there is an using strtol()
+#include <stdio.h>   /// for input, output
+#include <stdlib.h>  /// for exit() - to exit the program
+#include <string.h>  /// for strlen()
+/**
+ * @brief Determines the nth Fibonacci term
+ * @param number - n in "nth term" and it can't be negative as well as zero
+ * @return nth term in unsigned type
+ * @warning
+ * Only till 47th and 48th fibonacci element can be stored in 'int' and
+ * 'unsigned' respectively (takes more than 20 seconds to print)
+ */
+unsigned int fib(int number)
 {
-    if (number == 1 || number == 2)
+    // Check for negative integers
+    if (number <= 0)
+    {
+        fprintf(stderr, "Illegal Argument Is Passed!\n");
+        exit(EXIT_FAILURE);
+    }
+
+    // Base conditions
+    if (number == 1)
+        return 0;
+
+    if (number == 2)
         return 1;
-    else
-        return fib(number - 1) + fib(number - 2);
+
+    // Recursive call to the function
+    return fib(number - 1) + fib(number - 2);
 }
 
+/**
+ * @brief Get the input from the user
+ * @return valid argument to the fibonacci function
+ */
+int getInput(void)
+{
+    int num, success = 0;
+    char buffer[4], *endPtr;
+
+    do
+    {
+        fgets(buffer, 4, stdin);  // Inputs the value from user
+
+        errno = 0;  // Reseting the errno value
+        num = strtol(buffer, &endPtr,
+                     10);  // Attempts to convert the string to integer
+
+        if (strlen(buffer) > 3 || num > 48)
+            success = 0;  // Three digit numbers are not allowed
```

Comment:
why do we use `success` val instead of using `continue` (success = 0) and `break` instead of success = 1?

Reply:
Then what will be the while condition